batch/batch/driver/instance_collection/pool.py

```diff
@@ -536,65 +536,49 @@ async def schedule_loop_body(self):
         SCHEDULING_LOOP_RUNS.labels(pool_name=self.pool.name).inc()
 
         n_scheduled = 0
-        n_attempted_to_schedule = 0
 
-        n_cores_mcpu_attempted_to_schedule = 0
-        n_cores_mcpu_scheduled = 0
+        scheduled_jobs_per_region = defaultdict(int)
+        scheduled_cores_mcpu_per_region = defaultdict(int)
+        unscheduled_jobs_per_region = defaultdict(int)
+        unscheduled_cores_mcpu_per_region = defaultdict(int)
 
         user_resources = await self.compute_fair_share()
 
         total = sum(resources['allocated_cores_mcpu'] for resources in user_resources.values())
         if not total:
             should_wait = True
+            for region in self.pool.all_supported_regions:
+                SCHEDULING_LOOP_CORES.labels(pool_name=self.pool.name, region=region, scheduled=True).set(0)
+                SCHEDULING_LOOP_CORES.labels(pool_name=self.pool.name, region=region, scheduled=False).set(0)
+                SCHEDULING_LOOP_JOBS.labels(pool_name=self.pool.name, region=region, scheduled=True).set(0)
+                SCHEDULING_LOOP_JOBS.labels(pool_name=self.pool.name, region=region, scheduled=False).set(0)
```

Comment:
I think this loop should be removed. These metrics get updated at the end of the loop with their actual values and if we get scraped between now and the prometheus metric will plummet which is not accurate.

Reply:
The issue is with the code here, the entire loop doesn't actually get run. We return early if there are no cores to schedule:

```python3
        total = sum(resources['allocated_cores_mcpu'] for resources in user_resources.values())
        if not total:
            should_wait = True
            for region in self.pool.all_supported_regions:
                SCHEDULING_LOOP_CORES.labels(pool_name=self.pool.name, region=region, scheduled=True).set(0)
                SCHEDULING_LOOP_CORES.labels(pool_name=self.pool.name, region=region, scheduled=False).set(0)
                SCHEDULING_LOOP_JOBS.labels(pool_name=self.pool.name, region=region, scheduled=True).set(0)
                SCHEDULING_LOOP_JOBS.labels(pool_name=self.pool.name, region=region, scheduled=False).set(0)
            return should_wait
```